init/CMakeLists.txt

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2007-2017 Hartmut Kaiser
+# Copyright (c)      2011 Bryce Lelbach
+# Copyright (c)      2018 Nikunj Gupta
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+set(hpx_init_HEADERS "" CACHE INTERNAL "Headers for libhpx_init." FORCE)
+list(APPEND hpx_init_HEADERS
+  hpx/hpx_main.hpp
+  hpx/hpx_main_impl.hpp)
+
+set(hpx_init_SOURCES "" CACHE INTERNAL "Sources for libhpx_init." FORCE)
+list(APPEND hpx_init_SOURCES
+  hpx_main.cpp
+  hpx_main_argc_argv.cpp
+  hpx_main_variables_map.cpp
+  hpx_main_winsocket.cpp
+  hpx_user_main.cpp
+  hpx_user_main_argc_argv.cpp
+  main.cpp)
+
+if(HPX_WITH_DYNAMIC_HPX_MAIN AND (("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux") OR (APPLE)))
```

Comment:
Should the `HPX_WITH_DYNAMIC_HPX_MAIN` option be defined in this CMakeLists file?

Reply:
Yeah, why not! I thought I'd leave it with all the other options, but I see no reason not to move it.